onnxruntime/python/tools/quantization/matmul_4bits_quantizer.py

```diff
@@ -13,26 +15,122 @@
+        """
+        self.algorithm = algorithm
+        self.accuracy_level = accuracy_level
+
+
+class RTNWeightOnlyQuantConfig(WeightOnlyQuantConfig):
+    def __init__(
+        self,
+        accuracy_level=0,
+        ratios=None,
+    ):
+        """
+        This is a class for round-to-nearest (RTN) algorithm Weight Only Quant Configuration.
+        RTN is the most straightforward way to quantize weight using scale maps.
+
+        Args:
+            accuracy_level:
+                support 0 (default fp32), 1 (optimized fp32 for intel CPU), 2 (fp16), 3 (bf16), 4 (int8). Set to 0 by default.
+            ratios:
+                percentile of clip. Defaults to {}.
+        """
+        if ratios is None:
+            ratios = {}
+        super().__init__(
+            algorithm="RTN",
+            accuracy_level=accuracy_level,
+        )
+        self.ratios = ratios
+
+
+class GPTQWeightOnlyQuantConfig(WeightOnlyQuantConfig):
+    def __init__(
+        self,
+        calibration_data_reader: CalibrationDataReader,
+        percdamp=0.01,
+        blocksize=128,
+        actorder=False,
+        mse=False,
+        perchannel=True,
+        accuracy_level=0,
+    ):
+        """
+        This is a class for GPTQ algorithm Weight Only Quant Configuration.
+        GPTQ algorithm provides more accurate quantization but requires more computational resources.
+
+        Args:
+            calibration_data_reader:
+                a calibration data reader. It enumerates calibration data and generates inputs for the original model.
+            percdamp:
+                percent of the average Hessian diagonal to use for dampening.
+            blocksize (int, optional):
+                channel number in one block to execute a GPTQ quantization iteration.
+            actorder (bool, optional):
+                whether rearrange Hessian matrix considering the diag's value.
+            mse (bool, optional):
+                whether get scale and zero point with mse error.
+            perchannel (bool, optional):
+                whether quantize weight per-channel.
+            accuracy_level:
+                support 0 (default fp32), 1 (optimized fp32 for intel CPU), 2 (fp16), 3 (bf16), 4 (int8). Set to 0 by default.
+        """
+        super().__init__(
+            algorithm="GPTQ",
+            accuracy_level=accuracy_level,
+        )
+        self.calibration_data_reader = calibration_data_reader
+        self.percdamp = percdamp
+        self.blocksize = blocksize
+        self.actorder = actorder
+        self.mse = mse
+        self.perchannel = perchannel
+
+
 class MatMul4BitsQuantizer:
     """Perform 4b quantization of constant MatMul weights"""
 
-    def __init__(self, model: ModelProto, block_size: int, is_symmetric: bool, nodes_to_exclude=None):
+    def __init__(
+        self,
+        model: ModelProto,
```

Comment:
for large model, do we need choose to use model_path instead of model instance? 

I got the following warnings. 
![image](https://github.com/microsoft/onnxruntime/assets/817030/24fb9fe6-d8eb-4d66-b84e-ab5ea26e514c)

Reply:
Do you mean "model: Union[ModelProto, str]", or "model: str" and initialize model instance in MatMul4BitsQuantizer?

Reply:
II mean model: str. From what I tried, it seems line 300 will call https://github.com/intel/neural-compressor/blob/master/neural_compressor/model/onnx_model.py#L51, which will raise the warning if the model_path is None.

Reply:
There are some calls of MatMul4BitsQuantizer in onnxruntime and its model parameter is model instance, maybe use Union is more compatible？

Reply:
yes, that's another option.